src/views/routes/form/ServiceSelector.tsx

```diff
@@ -55,44 +56,71 @@ const ServiceSelector: FC<ServiceSelectorProps> = ({ namespace }) => {
-            <Select
-              id={SERVICE_FIELD_ID}
-              selected={value}
-              toggleContent={
-                value ? (
-                  <>
-                    <ResourceIcon groupVersionKind={ServiceGroupVersionKind} /> {value}
-                  </>
-                ) : (
-                  t('Select a Service')
-                )
-              }
-            >
-              <>
-                {!loaded && <Loading />}
-                {loaded &&
-                  (services || []).map((service) => (
-                    <DropdownItem
-                      key={getName(service)}
-                      onClick={() => {
-                        onChange(getName(service));
-                        setValue('spec.port.targetPort', '');
-                      }}
-                      value={getName(service)}
-                    >
-                      <ResourceIcon groupVersionKind={ServiceGroupVersionKind} /> {getName(service)}
-                    </DropdownItem>
-                  ))}
-              </>
-            </Select>
+          <>
+            <FormGroup fieldId={SERVICE_FIELD_ID} isRequired label={t('Service')}>
+              <Select
+                id={SERVICE_FIELD_ID}
+                selected={value}
+                toggleContent={
+                  value ? (
+                    <>
+                      <ResourceIcon groupVersionKind={ServiceGroupVersionKind} /> {value}
+                    </>
+                  ) : (
+                    t('Select a Service')
+                  )
+                }
+              >
+                <>
+                  {!loaded && <Loading />}
+                  {loaded &&
+                    (services || []).map((service) => (
+                      <DropdownItem
+                        key={getName(service)}
+                        onClick={() => {
+                          onChange(getName(service));
+                          setValue('spec.port.targetPort', '');
+                        }}
+                        value={getName(service)}
+                      >
+                        <ResourceIcon groupVersionKind={ServiceGroupVersionKind} />{' '}
+                        {getName(service)}
+                      </DropdownItem>
+                    ))}
+                </>
+              </Select>
 
-            <FormHelperText>
-              <HelperText>
-                <HelperTextItem>{t('Service to route to.')}</HelperTextItem>
-              </HelperText>
-            </FormHelperText>
-          </FormGroup>
+              <FormHelperText>
+                <HelperText>
+                  <HelperTextItem>{t('Service to route to.')}</HelperTextItem>
+                </HelperText>
+              </FormHelperText>
+            </FormGroup>
+
+            <FormGroup fieldId={SERVICE_WEIGHT_FIELD_ID} label={t('Service weight')}>
+              <TextInput
+                defaultValue={100}
```

Comment:
why do we want to give this arbitrary default value? if there's a good reason, add this to a const with a meaningful name

Reply:
okay. if this field is not defined, the default value is 100. this is the reason. I'll add a const